server/api/api/endpoints/model_endpoints.py

```diff
@@ -343,16 +357,113 @@ async def get_model_endpoint_monitoring_metrics(
             project=project
         ).get_model_endpoint_metrics,
         endpoint_id=endpoint_id,
     )
+
+
+@dataclass
+class _MetricsValuesParams:
+    project: str
+    endpoint_id: str
+    metrics: list[
+        mlrun.common.schemas.model_monitoring.model_endpoints.ModelEndpointMonitoringMetric
+    ]
+    start: datetime
+    end: datetime
+
+
+class _MetricsValuesInput(BaseModel):
+    name: conlist(
+        constr(
+            regex=mlrun.common.schemas.model_monitoring.model_endpoints._FQN_PATTERN
+        ),
+        min_items=1,
+    )  # pyright: ignore[reportInvalidTypeForm]
+    start: Optional[datetime] = None
+    end: Optional[datetime] = None
+
+
+async def _get_metrics_values_data(
+    project: str,
+    endpoint_id: str,
+    inputs: _MetricsValuesInput,
+    auth_info: mlrun.common.schemas.AuthInfo = Depends(
+        server.api.api.deps.authenticate_request
+    ),
+) -> _MetricsValuesParams:
+    """
+    Verify authorization, validate parameters and initialize the parameters.
+
+    :param project:     The name of the project.
+    :param endpoint_id: The unique id of the model endpoint.
+    :param inputs:      The full names of the requested results. At least one is required.
+                        Start and end times are optional, and must be timezone aware.
+    :param auth_info:   The auth info of the request.
+
+    :return:            _MetricsValuesData object with the validated data.
+    """
+    await _verify_model_endpoint_read_permission(
+        project=project, endpoint_id=endpoint_id, auth_info=auth_info
+    )
+    if inputs.start is None and inputs.end is None:
+        end = mlrun.utils.helpers.datetime_now()
+        start = end - timedelta(days=1)
+    elif inputs.start is not None and inputs.end is not None:
+        if inputs.start.tzinfo is None or inputs.end.tzinfo is None:
+            raise mlrun.errors.MLRunInvalidArgumentTypeError(
+                "Custom start and end times must contain the timezone."
+            )
+        if inputs.start > inputs.end:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "The start time must precede the end time."
+            )
+    else:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            "Provided only one of start time, end time. Provide both or none of them"
+        )
+    metrics = [
+        mlrun.common.schemas.model_monitoring.model_endpoints._parse_metric_fqn_to_monitoring_metric(
+            fqn
+        )
+        for fqn in inputs.name
+    ]
+    return _MetricsValuesParams(
+        project=project,
+        endpoint_id=endpoint_id,
+        metrics=metrics,
+        start=start,
+        end=end,
+    )
+
+
+@router.get(
+    "/{endpoint_id}/metrics-values",
+    response_model=list[
+        mlrun.common.schemas.model_monitoring.model_endpoints._ModelEndpointMonitoringResultValuesBase
+    ],
+)
+async def get_model_endpoint_monitoring_metrics_values(
```

Comment:
Shouldn't we add an integration test for this new API?

Reply:
Done in 24778da.